huaweicloud/services/rds/resource_huaweicloud_rds_pg_hba.go

```diff
@@ -0,0 +1,245 @@
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the connection type.`,
+			},
+			"database": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the database name.`,
+			},
+			"user": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the Name of a user.`,
+			},
+			"address": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the client IP address.`,
+			},
+			"method": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the authentication mode.`,
+			},
+			"mask": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the subnet mask.`,
+			},
+		},
+	}
+	return &sc
+}
+
+func resourcePgHbaCreateOrUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	err := setPgHba(d, meta, buildPgHbaBodyParams(d))
+	if err != nil {
+		return diag.Errorf("error setting RDS PostgreSQL hba: %s", err)
+	}
+
+	d.SetId(d.Get("instance_id").(string))
+
+	return resourcePgHbaRead(ctx, d, meta)
+}
+
+func buildPgHbaBodyParams(rawParams interface{}) []map[string]interface{} {
+	if rawArray, ok := rawParams.([]interface{}); ok {
+		if len(rawArray) == 0 {
+			return nil
+		}
+
+		rst := make([]map[string]interface{}, len(rawArray))
+		for i, v := range rawArray {
+			raw := v.(map[string]interface{})
+			rst[i] = map[string]interface{}{
+				"priority": i,
+				"type":     raw["type"],
+				"database": raw["database"],
+				"user":     raw["user"],
+				"address":  raw["address"],
+				"method":   raw["method"],
+				"mask":     utils.ValueIngoreEmpty(raw["mask"]),
+			}
+		}
+		return rst
+	}
+	return nil
+}
+
+func resourcePgHbaRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getPgHba: query RDS PostgreSQL hba
+	var (
+		getPgHbaHttpUrl = "v3/{project_id}/instances/{instance_id}/hba-info"
+		getPgHbaProduct = "rds"
+	)
+	getPgHbaClient, err := cfg.NewServiceClient(getPgHbaProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating RDS client: %s", err)
+	}
+
+	getPgHbaPath := getPgHbaClient.Endpoint + getPgHbaHttpUrl
+	getPgHbaPath = strings.ReplaceAll(getPgHbaPath, "{project_id}", getPgHbaClient.ProjectID)
+	getPgHbaPath = strings.ReplaceAll(getPgHbaPath, "{instance_id}", d.Id())
+
+	getPgHbaOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+	}
+
+	getPgHbaResp, err := getPgHbaClient.Request("GET", getPgHbaPath, &getPgHbaOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving RDS PostgreSQL hba")
+	}
+
+	getPgHbaRespBody, err := utils.FlattenResponse(getPgHbaResp)
```

Comment:
- When all the returned lists are default hosts, `404` error code should be thrown.
- When the return body contains `"errCode": "DBS.280343"`,  `404` error code should be thrown.

Reply:
the errCode of "DBS.280343" has done, the resource should not be destroyed when the rules are all default rules, because the value can be set to the default by user